tests/test_algebra_custom_model.py

```diff
@@ -0,0 +1,145 @@
+"""
+Tests scikit-learn's binarizer converter.
+"""
+import unittest
+import numpy as np
+from distutils.version import StrictVersion
+import onnx
+import onnx.checker
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from skl2onnx.algebra.onnx_operator_mixin import OnnxOperatorMixin
+from skl2onnx import convert_sklearn
+from skl2onnx.convert import to_onnx
+from skl2onnx.common.data_types import FloatTensorType
+from skl2onnx.algebra.onnx_ops import OnnxDiv, OnnxSub
+from test_utils import dump_data_and_model
+
+
+class CustomOpTransformer(BaseEstimator, TransformerMixin,
+                          OnnxOperatorMixin):
+
+    def __init__(self):
+        BaseEstimator.__init__(self)
+        TransformerMixin.__init__(self)
+
+    def fit(self, X, y=None):
+        self.W_ = np.mean(X, axis=0)
+        self.S_ = np.std(X, axis=0)
+        return self
+
+    def transform(self, X):
+        return (X - self.W_) / self.S_
+
+    def to_onnx_operator(self, inputs=None, outputs=('Y', )):
+        if inputs is None:
+            raise RuntimeError("inputs should contain one name")
+        i0 = self.get_inputs(inputs, 0)
+        W = self.W_
+        S = self.S_
+        # case if there are multiple output nodes
+
+        return OnnxDiv(OnnxSub(i0, W), S,
+                       output_names=outputs)
+
+
+class CustomOpTransformerShape(CustomOpTransformer):
+    def onnx_shape_calculator(self):
+        def shape_calculator(operator):
+            operator.outputs[0].type = FloatTensorType(
+                shape=operator.inputs[0].type.shape)
+        return shape_calculator
+
+
+class TestCustomModelAlgebra(unittest.TestCase):
+
+    def test_base_api(self):
+
```

Comment:
Why do you have empty lines at the start of method definitions in this class?

Reply:
Fixed.